### ISSUE TITLE
Elyra 2.0 install from PyPI fails due to conflicting dependencies

### ISSUE THREAD

**Issue by ptitzler** [elyra-ai/elyra#1251]
**Describe the issue**

```
ERROR: Cannot install elyra and elyra==2.0.0 because these package versions have conflicting dependencies.
The conflict is caused by:
    elyra 2.0.0 depends on kfp==1.1.2
    kfp-tekton 0.5.0 depends on kfp==1.0.4
```

**To Reproduce**
1. In a clean environment run  `pip install --upgrade "elyra>=2.0.0" && jupyter lab build`

**Expected behavior**
Elyra 2.0 is installed.

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 2.0.0
 - Operating system: macos
 - Installation source: PyPI
 - Deployment type: local installation

**Comment by lresende** [comment]
This should only happen if you have the latest pip installed. 
While we work on sorting this out on the dependency list, a workaround would be to use
```
pip install --upgrade --use-deprecated=legacy-resolver elyra && jupyter lab build
```

**Comment by akchinSTC** [comment]
kfp-tekton is planning a release next week where its kfp requirement will be 1.3.0, kfp==1.3.0. We would need to relax our kfp requirements for the default resolver to work. e.g. >1.0.0, <=1.3.0
Running tests now to find the floor

**Comment by lresende** [comment]
It actually looks like we need to take care of two things:
```
kfp-tekton 0.5.0 requires kfp==1.0.4, but you'll have kfp 1.1.2 which is incompatible.
papermill 2.3.1 requires nbformat>=5.1.2, but you'll have nbformat 5.0.8 which is incompatible.
```

**Comment by ckadner** [comment]
I just uploaded an unofficial `kfp-tekton` package to PyPI, version `0.5.1rc0` (https://pypi.org/project/kfp-tekton/0.5.1rc0/) which is aligned with `kfp==1.1.2`

Can you give that a try?

```shell
pip install kfp-tekton==0.5.1rc0
```

**Comment by lresende** [comment]
Thank you @ckadner, will give it a try shortly.

**Comment by ckadner** [comment]
I just pushed another RC (`rc1`) to workaround a `pip` install error on mismatching version stamps `0.5.1  != 0.5.1rc0`

```shell
pip install kfp-tekton==0.5.1rc1
```

**Comment by lresende** [comment]
Fixed by #1254 and #1252 with final move to kfp-tekton 0.5.1rc1
